src/youtube.js

```diff
@@ -0,0 +1,1094 @@
+    if (this._api) {
+      if (this.currentTime === this.duration) {
+        this.currentTime = 0;
+      }
+      this._api.playVideo();
+    }
+  }
+
+  /**
+   * Pause playback.
+   * @public
+   * @returns {void}
+   */
+  pause(): void {
+    if (this._api) {
+      this._api.pauseVideo();
+    }
+  }
+
+  /**
+   * Load media.
+   * @param {number} startTime - Optional time to start the video from.
+   * @public
+   * @returns {Promise<Object>} - The loaded data
+   */
+  load(startTime: ?number): Promise<Object> {
+    this._loaded = true;
+    this._sdkLoaded
+      .then(() => {
+        //TODO: if autoplay pass it here and check in play if already playing?
+        const loadOptions = {
+          videoId: this._source.url,
+          startSeconds: (startTime && startTime > 0) ? startTime : 0
+        };
+        this._api.loadVideoById(loadOptions);
+      })
+      .catch(error => {
+        return Promise.reject(error);
+      });
+    this._videoLoaded = {};
+    this._videoLoaded.promise = new Promise((resolve, reject) => {
+      this._videoLoaded.resolve = resolve;
+      this._videoLoaded.reject = reject;
+    });
+    return this._videoLoaded.promise;
+  }
+
+  /**
+   * Parse hls video tracks into player video tracks.
+   * @returns {Array<VideoTrack>} - The parsed video tracks.
+   * @private
+   */
+  _parseVideoTracks(): Array<VideoTrack>{
+    const levels: Array<string> = this._api.getAvailableQualityLevels();
+    const currentLevel: string = this._api.getPlaybackQuality();
+    const videoTracks = [];
+    levels.forEach((level, i) => {
+      let settings = {
+        active: currentLevel === level,
+        label: level,
+        bandwidth: 0,
+        width: i+1, //TODO: get estimated width and height from https://developers.google.com/youtube/iframe_api_reference#Playback_quality
+        height: i+1,
+        language: '',
+        index: i
+      };
+      videoTracks.push(new VideoTrack(settings));
+    });
+    return videoTracks;
+  }
+
+  /**
+   * Set a source.
+   * @param {string} source - Source to set.
+   * @public
+   * @returns {void}
+   */
+  set src(source: string): void {
+    this._source.url = source;
+  }
+
+  /**
+   * Get the source url.
+   * @returns {string} - The source url.
+   * @public
+   */
+  get src(): string {
+    return (this._loaded && this._source) ? this._source.url: "";
+  }
+
+  /**
+   * Get the current time in seconds.
+   * @returns {Number} - The current playback time.
+   * @public
+   */
+  get currentTime(): number {
+    return this._api ? this._api.getCurrentTime() : 0;
+  }
+
+  /**
+   * Set the current time in seconds.
+   * @param {Number} to - The number to set in seconds.
+   * @public
+   * @returns {void}
+   */
+  set currentTime(to: number): void {
+    if (this._api) {
+      if (this.currentTime !== to) {
+        this._isSeeking = true;
+        this._api.seekTo(to, true);
+        this.dispatchEvent(new FakeEvent(EventType.SEEKING));
```

Comment:
what's about `SEEKED`? triggered from the api?

Reply:
No, they don't handle seeking or seeked at all